python/lancedb/table.py

```diff
@@ -524,22 +523,90 @@ def update(
 class LanceTable(Table):
     """
     A table in a LanceDB database.
+
+    This can be opened in two modes: if `version` is None then the latest version
+    is opened and can be tracked. If `version` is an integer then that version is
+    opened and this instance is fixed to that version.
     """
 
     def __init__(
-        self, connection: "lancedb.db.LanceDBConnection", name: str, version: int = None
+        self,
+        connection: "lancedb.db.LanceDBConnection",
+        name: str,
+        version: Optional[int] = None,
+        /,
+        read_consistency_interval: Optional[timedelta] = None,
     ):
         self._conn = connection
         self.name = name
-        self._version = version
+        # To be lazy, we initially store the desired version as a LanceDataset
+        self._version: Optional[int] = version
+        self._ds: Optional[LanceDataset] = version
 
-    def _reset_dataset(self, version=None):
-        try:
-            if "_dataset" in self.__dict__:
-                del self.__dict__["_dataset"]
-            self._version = version
-        except AttributeError:
-            pass
+        if self._version is not None and self.read_consistency_interval is not None:
+            raise ValueError(
+                "Cannot specify both version and read_consistency_interval."
+            )
+        self.read_consistency_interval = read_consistency_interval
+        self._last_consistency_check = None
+
+    @classmethod
+    def open(cls, db, name, **kwargs):
+        tbl = cls(db, name, **kwargs)
+        fs, path = fs_from_uri(tbl._dataset_uri)
+        file_info = fs.get_file_info(path)
+        if file_info.type != pa.fs.FileType.Directory:
+            raise FileNotFoundError(
+                f"Table {name} does not exist."
+                f"Please first call db.create_table({name}, data)"
+            )
+        register_event("open_table")
```

Comment:
What does `register_event` do?

Reply:
That's the event tracking we setup to collect analytics on usage.